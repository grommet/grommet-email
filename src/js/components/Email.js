// (C) Copyright 2017 Hewlett Packard Enterprise Development LP

import React, { PropTypes } from 'react';
import { buildStyle, fontFace } from '../utils/Style';

const Email = (props) => {
  const style = buildStyle(['fontFamily', 'color', 'fontSize', 'lineHeight']);
  return (
    <html lang="en-US">
      <head>
        <style type="text/css">
          {fontFace}
        </style>
      </head>
      <body style={style}>
        {props.children}
      </body>
    </html>
  );
};

Email.propTypes = {
  children: PropTypes.any,
};

export default Email;
