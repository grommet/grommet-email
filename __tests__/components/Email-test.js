// (C) Copyright 2017 Hewlett Packard Enterprise Development LP

import React from 'react';
import renderer from 'react-test-renderer';

import Email from '../../src/js/components/Email';

describe('Email', () => {
  it('has correct default options', () => {
    const component = renderer.create(
      <Email>Testing</Email>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
