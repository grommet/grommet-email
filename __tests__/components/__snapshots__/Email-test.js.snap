// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Email has correct default options 1`] = `
<html
  lang="en-US"
>
  <head>
    <style
      type="text/css"
    />
  </head>
  <body
    style={
      Object {
        "color": "#666",
        "fontFamily": "Metric, Arial, sans-serif",
        "fontSize": undefined,
        "lineHeight": 24,
      }
    }
  >
    Testing
  </body>
</html>
`;
